mono/mini/interp/interp.c

```diff
@@ -4490,6 +4515,30 @@ interp_exec_method_full (InterpFrame *frame, ThreadContext *context, guint16 *st
 			++sp;
 			MINT_IN_BREAK;
 		}
+
+#define STSFLD(datamem, fieldtype) { \
+	MonoVTable *vtable = (MonoVTable*) rtm->data_items [*(guint16*)(ip + 1)]; \
```

Comment:
This `*(guint16*)(ip + 1)` thing can't alignment fault?
We have forms of that, that do the same for x86/amd64, but are more careful otherwise?

Reply:
we use that access pattern everywhere in the interpreter

Reply:
Wierd. See `NO_UNALIGNED_ACCESS`.

Reply:
Oh, extra wierd, `NO_UNALIGNED_ACCESS` is only about 32bit and 64bit operations, not 16b.t

Reply:
ip is defined as short*.

Reply:
I didn't realize that, though that doesn't make it is aligned per se. It means if it is aligned, that +1 will still be aligned. I think it comes down to this being the interpreter IR, composed of 16bit opcodes, and not the original ECMA IL of 8bit opcodes, so ok. Which would also guarantee native endianess.